### DISCUSSION TITLE
Pandas-like plot shortcuts

### DISCUSSION THREAD

**Discussion by henryiii** [scikit-hep/hist#294] [Ideas]
Pandas has several useful keyword arguments that wrap mpl functionality. We could make wrapper that provides this for all our functions. For example, for legend (also requested for at lest logy):

```python
# For None, the check for get_legend_handles_labels would not throw an error, but just skip
def _plot_keywords(ax: Axis, legend: bool | None) -> None:
    "Anything added here must also be in pandas"

    if legend:
        if ax.get_legend_handles_labels()[0]:
             legend = ax.legend()
        else:
            raise <some error>

...
# For Stack, this defaults to legend=None
def plot(self, *, legend: bool | None = False, **kwargs: Any) -> Any:
        ...
        
       ret: Artist = ...
       ax = ret[0].ax
       hist.plot._plot_keywords(ax, legend=legend)
```

This is convenience only, this can be done by hand with a few lines. If we do add any of these keywords, this is how we should do it. Do we add this, though? Or keep to base mpl commands? _The first step_ is making sure we have good examples of the base commands in our docs.

**Comment by LovelyBuggies** [comment]
@henryiii FYI, `ax = ret[0].ax` doesn't work (neither "axis" or "axes" works). Info about `ret` (len=3):

```
[StairsArtists(stairs=<matplotlib.patches.StepPatch object at 0x1842fbd90>, errorbar=<ErrorbarContainer object of 3 artists>, legend_artist=<ErrorbarContainer object of 3 artists>), StairsArtists(stairs=<matplotlib.patches.StepPatch object at 0x184342a60>, errorbar=<ErrorbarContainer object of 3 artists>, legend_artist=<ErrorbarContainer object of 3 artists>), StairsArtists(stairs=<matplotlib.patches.StepPatch object at 0x184363760>, errorbar=<ErrorbarContainer object of 3 artists>, legend_artist=<ErrorbarContainer object of 3 artists>)]
```

Wonder why there are 3 `StairsArtists` and how to obtain the axis.

**Reply by henryiii** [reply]
Looks like it's:

```python
a[0][0].axes
# OR
a[0].stairs.axes
# OR
a[0].legend_artist[0].axes
```

The StairsArtist has several parts, including an error bar and a legend artist, so you need to get one of those (like the stairs itself) to get the axes.

**Reply by henryiii** [reply]
(I used `dir(a[0])` to see what was allowed)
